### DISCUSSION TITLE
How to set contact photo?

### DISCUSSION THREAD

**Discussion by amoshalava** [vestrel00/contacts-android#170] [Q&A] [answered]
val contact = insertResult.contacts(contactsApi).firstOrNull()

I keep getting error on contactsApi. I'm not sure I'm using it correctly. I can't figure out what to set it to. I can pull pictures of my contacts from an API URL and want to push that photo into the contact.

**Comment by vestrel00** [comment, marked as answer]
> I keep getting error on contactsApi.

Can you tell me what the error is? Or is it just not working the way you want it to?

> I can pull pictures of my contacts from an API URL and want to push that photo into the contact.

I see. So, if you want to add pictures to contacts you create, here is the how-to page for that; https://github.com/vestrel00/contacts-android/blob/main/howto/howto-get-set-remove-contact-raw-contact-photo.md#setting-contact-photo

Currently, the library only supports adding photos after the contact has been inserted. However, this will change in #119 allowing you to include photos in an insert call.

You can still set photos using the library now. It would look something like this (assuming that your call to insert the contact succeeds);

```kotlin
val contact = insertResult.contacts(contactsApi).firstOrNull()

if (contact != null) {
    contact.setPhoto(contactsApi, photo)
} else {
    // Let @vestrel00 know if this is happening.
}
```

The `photo` can be anyone of these; `ByteArray`, `InputStream`, `Bitmap`, and `BitmapDrawable`. So if your photos are not in one of those formats, you will have to convert them first. Image conversion/manipulation are not in scope of this library so I suggest looking at other imaging libraries if you need to convert your photos.

Let me know how it goes!

**Reply by amoshalava** [reply]
where do you set or define contactsApi?

**Reply by vestrel00** [reply]
It's the same `contactsApi` that you use to query and insert contacts or you can create a new one `Contacts(context)`
